src/shogun/evaluation/MachineEvaluation.cpp

```diff
@@ -121,5 +165,19 @@ CMachine* CMachineEvaluation::get_machine() const
 
 EEvaluationDirection CMachineEvaluation::get_evaluation_direction()
 {
-	return m_evaluation_criterion->get_evaluation_direction();
+    REQUIRE(m_list_evaluation_criterion->get_array_size()==1,"Multiple Metrics provided. Please use get_evaluation_directions()");
+    CEvaluation* temp_eval = (CEvaluation*) m_list_evaluation_criterion->get_element_safe(0);
+    return temp_eval->get_evaluation_direction();
+}
+
+CDynamicArray<EEvaluationDirection> CMachineEvaluation::get_evaluation_directions()
+{
+    CDynamicArray<EEvaluationDirection> list_evaluation_direction = *(new CDynamicArray<EEvaluationDirection>());
```

Comment:
Thanks for pointing it out. I understand the logic. I did it because, the corresponding method for a single metric returned by value. If this is the case, I might have to modify the earlier method. Also the method [`CEvaluation::get_evaluation_direction()`](http://www.shogun-toolbox.org/doc/en/latest/classshogun_1_1CEvaluation.html#a3b8baece613508b24b1a0bd645e507c5) itself returns by value of which I am making a list and returning. So, even if I return a pointer, no future changes are going to be reflected.